### PR TITLE
fix(vector sink): cert verification with proxy enabled

### PR DESCRIPTION
**Pull request by joemiller**
fixes https://github.com/vectordotdev/vector/issues/17629

### CODE REVIEW

File: src/sinks/vector/config.rs [hunk 1]
```diff
@@ -224,6 +224,9 @@ fn new_client(
     });
 
     let mut proxy = ProxyConnector::new(https).unwrap();
+    // Make proxy connector aware of user TLS settings by setting the TLS connector:
```
Comment (neuronull) [inline src/sinks/vector/config.rs:227]:
```suggestion
    // Make proxy connector aware of user TLS settings by setting the TLS connector.
```

I think the ":" was reference to the GH issue linked in `src/http.rs`

**Review by neuronull — DISMISSED** [review]
Thanks for the contribution!

Would you mind adding the same comment we have above the tls builder (above line 214), as was added in `src/http.rs` ?

```
// Create dedicated TLS connector for the proxied connection with user TLS settings.
```

File: src/sinks/vector/config.rs (hunk 1 above)
Comment (neuronull) [reply src/sinks/vector/config.rs:227]: ignore this... see my new comment

**Comment by neuronull** [comment]
> Thanks for the contribution!
> 
> Would you mind adding the same comment we have above the tls builder (above line 214), as was added in `src/http.rs` ?
> 
> ```
> // Create dedicated TLS connector for the proxied connection with user TLS settings.
> ```

Actually, on further inspection, I believe this PR should just utilize `build_proxy_connector()` directly,  similar to what is being done here: https://github.com/vectordotdev/vector/blob/ee480cd08a5451bc3f0b83a2b037ba131e38d4b9/src/aws/mod.rs#L99-L100

**Comment by joemiller** [comment]
@neuronull This might be a bit outside my skillset at the moment. The 3 lines of rust in this PR were the first I have ever written.

**Comment by neuronull** [comment]
> @neuronull This might be a bit outside my skillset at the moment. The 3 lines of rust in this PR were the first I have ever written.

Hi @joemiller , I pushed up the suggested change as it was small.
Once https://github.com/vectordotdev/vector/pull/17841 is merged in, I will re-run CI.

**Review by spencergilbert — APPROVED** [review]

**Review by StephenWakely — DISMISSED** [review]
This code above where we are building the proxy connector is no longer being used and can be removed:

```rust
    let tls = tls_connector_builder(tls_settings)?;
    let mut https = HttpsConnector::with_connector(http, tls)?;
    let settings = tls_settings.tls().cloned();
    https.set_callback(move |c, _uri| {
        if let Some(settings) = &settings {
            settings.apply_connect_configuration(c);
        }
        Ok(())
    });
```

**Comment by neuronull** [comment]
> This code above where we are building the proxy connector is no longer being used and can be removed:

Good catch! Funny the compiler didn't complain because the variables used each other.

**Pull request merged**